File: backend/codeblurb-backend/src/main/java/hu/codeblurb/backend/controller/dto/QuizContentResponse.java
package hu.codeblurb.backend.controller.dto;

import lombok.Getter;

import java.util.List;

import static hu.codeblurb.backend.domain.content.Content.ContentType.QUIZ;

@Getter
public class QuizContentResponse extends ContentResponse {

    private final Integer id;
    private final List<QuizQuestionResponse> questions;

    public QuizContentResponse(Integer id, String name, List<QuizQuestionResponse> questions) {
        super(name, QUIZ);
        this.id = id;
        this.questions = questions;
    }

    @Override
    public String toString() {
        return "QuizContentResponse[" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", questions=" + questions +
                ']';
    }

}

File: mobile/codeblurb/lib/data/content/models/quiz_content_response.dart
import 'package:codeblurb/data/common/network/models/content_type.dart';
import 'package:codeblurb/data/content/models/quiz_question_response.dart';
import 'package:json_annotation/json_annotation.dart';

part 'quiz_content_response.g.dart';

@JsonSerializable(createToJson: false)
class QuizContentResponse {
  final String name;
  final int id;
  final List<QuizQuestionResponse> questions;

  QuizContentResponse(this.id, this.name, this.questions);

  ContentType get contentType => ContentType.quiz;

  factory QuizContentResponse.fromJson(Map<String, dynamic> json) =>
      _$QuizContentResponseFromJson(json);
}
